src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -3357,14 +3357,33 @@ bool Unit::IsUnderWater() const
     return GetBaseMap()->IsUnderWater(GetPositionX(), GetPositionY(), GetPositionZ());
 }
 
-void Unit::UpdateUnderwaterState(Map* m, float x, float y, float z)
+void Unit::UpdatePositionData()
+{
+    PositionFullVMapData data;
+    GetMap()->getFullVMapDataForPosition(GetPositionX(), GetPositionY(), GetPositionZ(), data);
+    
+    m_zoneId = m_areaId = data.areaInfo.areaId;
+    if (AreaTableEntry const* area = sAreaTableStore.LookupEntry(m_areaId))
+        if (area->zone)
+            m_zoneId = area->zone;
+
+    /*if (GetTypeId() == TYPEID_PLAYER)
```

Comment:
Heads-up, enable this block if you're reproing to get me some useful info (also, give me a `.gps`).